src/Explorer/Tabs/CloudShellTab/CloudShellTerminalCore.tsx

```diff
@@ -0,0 +1,274 @@
+      },
+      startKey,
+    );
+
+    return socket;
+  } catch (err) {
+    TelemetryProcessor.traceFailure(
+      Action.CloudShellTerminalSession,
+      {
+        shellType: TerminalKind[shellType],
+        dataExplorerArea: Areas.CloudShell,
+        region: resolvedRegion,
+        error: getErrorMessage(err),
+        errorStack: getErrorStack(err),
+      },
+      startKey,
+    );
+
+    terminal.writeln(formatErrorMessage(`Failed with error.${getErrorMessage(err)}`));
+
+    return null;
+  }
+};
+
+/**
+ * Ensures that the CloudShell provider is registered for the current subscription
+ */
+export const ensureCloudShellProviderRegistered = async (): Promise<void> => {
+  const response: CloudShellProviderInfo = await verifyCloudShellProviderRegistration(userContext.subscriptionId);
+
+  if (response.registrationState !== "Registered") {
+    await registerCloudShellProvider(userContext.subscriptionId);
+  }
+};
+
+/**
+ * Determines the appropriate CloudShell region
+ */
+export const determineCloudShellRegion = (): string => {
+  return getNormalizedRegion(userContext.databaseAccount?.location, DEFAULT_CLOUDSHELL_REGION);
+};
+
+/**
+ * Provisions a CloudShell session
+ */
+export const provisionCloudShellSession = async (
+  resolvedRegion: string,
+  terminal: Terminal,
+): Promise<{ socketUri?: string; provisionConsoleResponse?: ProvisionConsoleResponse; targetUri?: string }> => {
+  // Apply user settings
+  await putEphemeralUserSettings(userContext.subscriptionId, resolvedRegion);
+
+  // Provision console
+  let provisionConsoleResponse;
+  let attemptCounter = 0;
+
+  do {
+    provisionConsoleResponse = await provisionConsole(resolvedRegion);
+    attemptCounter++;
+
+    if (provisionConsoleResponse.properties.provisioningState === "Failed") {
+      break;
+    }
+
+    if (provisionConsoleResponse.properties.provisioningState !== "Succeeded") {
+      await wait(POLLING_INTERVAL_MS);
+    }
+  } while (provisionConsoleResponse.properties.provisioningState !== "Succeeded" && attemptCounter < MAX_RETRY_COUNT);
+
+  if (provisionConsoleResponse.properties.provisioningState !== "Succeeded") {
+    throw new Error(`Provisioning failed: ${provisionConsoleResponse.properties.provisioningState}`);
+  }
+
+  // Connect terminal
+  const connectTerminalResponse = await connectTerminal(provisionConsoleResponse.properties.uri, {
+    rows: terminal.rows,
+    cols: terminal.cols,
+  });
+
+  const targetUri = `${provisionConsoleResponse.properties.uri}/terminals?cols=${terminal.cols}&rows=${terminal.rows}&version=2019-01-01&shell=bash`;
+  const termId = connectTerminalResponse.id;
+
+  // Determine socket URI
+  let socketUri = connectTerminalResponse.socketUri.replace(":443/", "");
+  const targetUriBody = targetUri.replace("https://", "").split("?")[0];
+
+  if (socketUri.indexOf(targetUriBody) === -1) {
+    socketUri = `wss://${targetUriBody}/${termId}`;
+  }
+
+  if (targetUriBody.includes("servicebus")) {
```

Comment:
Please add a comment explaining when this occurs and why we need to modify the URI.